packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -105,16 +106,34 @@ export class DataSourceWithBackend<
         requestId,
       })
       .then((rsp: any) => {
-        return toDataQueryResponse(rsp);
+        const dqs = toDataQueryResponse(rsp);
+        if (this.processResponse) {
+          return this.processResponse(dqs);
+        }
+        return dqs;
       })
       .catch(err => {
         err.isHandled = true; // Avoid extra popup warning
-        return toDataQueryResponse(err);
+        const dqs = toDataQueryResponse(err);
+        if (this.processResponse) {
+          return this.processResponse(dqs);
+        }
+        return dqs;
```

Comment:
Nit: if we can it would be great to change this from Promise based chain into a await/async based chain.  But no biggie.